flask_resources/resources.py

```diff
@@ -1,153 +1,196 @@
 # -*- coding: utf-8 -*-
 #
-# Copyright (C) 2020 CERN.
-# Copyright (C) 2020 Northwestern University.
+# Copyright (C) 2020-2021 CERN.
+# Copyright (C) 2020-2021 Northwestern University.
 #
 # Flask-Resources is free software; you can redistribute it and/or modify it
 # under the terms of the MIT License; see LICENSE file for more details.
 
 """Resource view."""
 
+from inspect import ismethod
+
 from flask import Blueprint
-from marshmallow import ValidationError
+from werkzeug.exceptions import HTTPException
 
+from .config import from_conf, resolve_from_conf
+from .content_negotiation import with_content_negotiation
+from .context import ResourceRequestCtx
 from .deserializers import JSONDeserializer
-from .errors import create_errormap_handler
-from .loaders import RequestLoader
-from .parsers import HeadersParser, URLArgsParser
-from .responses import Response
+from .errors import handle_http_exception
+from .parsers import RequestBodyParser
+from .responses import ResponseHandler
 from .serializers import JSONSerializer
-from .views import ItemView, ListView, SingletonView
 
-ITEM_VIEW_SUFFIX = "_item"
-LIST_VIEW_SUFFIX = "_list"
+
+def route(
+    method,
+    rule,
+    view_meth,
+    endpoint=None,
+    rule_options={},
+    apply_decorators=True,
+):
+    """Create a route.
+
+    Use this method in ``create_url_rules()`` to build your list of rules.
+
+    The ``view_method`` parameter should be a bound method (e.g.
+    ``self.myview``).
+
+    :param method: The HTTP method for this URL rule.
+    :param rule: A URL rule.
+    :param view_meth: The view method (a bound method) for this URL rule.
+    :param endpoint: The name of the endpoint. By default the name is taken
+        from the method name.
+    :param rule_options: A dictionary of extra options passed to
+        ``Blueprint.add_url_rule``.
+    :param apply_decorators: Apply the decorators defined by the resource.
+        Defaults to ``True``. This allows you to selective disable
+        decorators which are normally applied to all view methods.
+    """
+    view_name = view_meth.__name__
+    config = view_meth.__self__.config
+    decorators = view_meth.__self__.decorators
+
+    if apply_decorators:
+        # reversed so order is the same as when applied directly to method
+        for decorator in reversed(decorators):
+            view_meth = decorator(view_meth)
+
+    def view(*args, **kwargs):
+        with ResourceRequestCtx(config):
```

Comment:
All resource methods run in the resource context, and the config is passed in from here.